nucypher/network/nodes.py

```diff
@@ -849,22 +849,19 @@ def learn_from_teacher_node(self, eager=False, canceller=None):
         #
         try:
             signature, node_payload = signature_splitter(response.content, return_remainder=True)
-        except BytestringSplittingError as e:
+        except BytestringSplittingError:
             self.log.warn("No signature prepended to Teacher {} payload: {}".format(current_teacher, response.content))
             return
 
         try:
             self.verify_from(current_teacher, node_payload, signature=signature)
-        except InvalidSignature as e:
-            self.suspicious_activities_witnessed['vladimirs'].append(
-                ('Node payload improperly signed', node_payload, signature))
-            self.log.warn(
-                f"Invalid signature ({signature}) received from teacher {current_teacher} for payload {node_payload}")
+        except Learner.InvalidSignature:  # TODO: Ensure wev've got the right InvalidSignature exception here
+            self.suspicious_activities_witnessed['vladimirs'].append(('Node payload improperly signed', node_payload, signature))
```

Comment:
We have:
* `Character.verify_from()` that raises `cryptography.exceptions.InvalidSignature`
* `Learner.verify_from()` that raises `Learner.InvalidSignature`
In the context of an object that is both a Character and a Learner, it seems extremely confusing.

Reply:
Yes, please open an issue for this.